src/WebApi/Infrastructure/Middleware/BasicAuthMiddleware.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using System.Net.Http.Headers;
+
+namespace Sindibad.SAD.WebTemplate.WebApi.Infrastructure.Middleware
+{
+    public class BasicAuthMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private const string Realm = "SindibadAPI";
+
+        public BasicAuthMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            if (!context.Request.Path.Value.Contains("api"))
+            {
+                await _next(context);
+                return;
+            }
+
+            if (!context.Request.Headers.ContainsKey("Authorization"))
+            {
+                context.Response.Headers["WWW-Authenticate"] = $"Basic realm=\"{Realm}\"";
+                context.Response.StatusCode = StatusCodes.Status403Forbidden;
+                await context.Response.WriteAsync("Access Forbidden.");
+                return;
+            }
+
+            var authHeader = AuthenticationHeaderValue.Parse(context.Request.Headers["Authorization"]);
+            var credentials = Encoding.UTF8.GetString(Convert.FromBase64String(authHeader.Parameter)).Split(':');
+            var username = credentials[0];
+            var password = credentials[1];
+
+            // Validate the username and password
+            if (!IsAuthorizedUser(username, password))
+            {
+                context.Response.StatusCode = StatusCodes.Status401Unauthorized;
+                await context.Response.WriteAsync("Invalid credentials.");
+                return;
+            }
+
+            await _next(context);
+        }
+
+        private bool IsAuthorizedUser(string username, string password)
+        {
+            return username == "Sindiboss" && password == "FindingSevenThievesInBGW!";
```

Comment:
It's true that we are implementing this only for internal use, it's still very possible to want to change these values often based on env or other concerns. It's probably best to keep them and read them from configuration (ie appsettings, env variables, etc.)